components/raftstore/src/store/fsm/peer.rs

```diff
@@ -5314,6 +5362,20 @@ where
         self.register_check_long_uncommitted_tick();
     }
 
+    fn on_request_snapshot_tick(&mut self) {
+        fail_point!("ignore request snapshot", |_| {});
+        assert!(!self.fsm.peer.is_leader());
+        if self
+            .fsm
+            .peer
+            .raft_group
+            .request_snapshot(self.fsm.peer.get_store().first_index())
+            .is_err()
```

Comment:
If it's the first time to call it, it should be not err. Then how do we keep ticking the tick?

Reply:
Changed no longer judges its return value, but registers a tick again (the current interval setting is 1min), the next time it is triggered, it will first check wait_data, if it is still true, repeat the above process, otherwise return directly.